### PR TITLE
Don't log to Slack on 'info'

### PR DESCRIPTION
**Pull request by diurnalist**
This refactors all of the hammers to only log to Slack when it actually did something important. The information messages still get printed out to stdout.

This required refactoring business logic in most if not all of the hammers, because the info vs. delete vs. update logic was strictly intertwined, as well as the branching of slack vs. not slack.

### CODE REVIEW

File: hammers/scripts/conflict_macs.py [hunk 1]
```diff
@@ -77,20 +77,36 @@ def main(argv=None):
...
-            show_info(conflict_macs, slack=slack)
+            show_info(conflict_macs)
         elif args.mode == 'delete':
-            delete(auth, conflict_macs,
-                safe=not args.force_sane, slack=slack, verbose=args.verbose)
+            if (not args.force_sane) and len(conflict_macs) > 10:
+                raise RuntimeError('(in)sanity check: thinks there are {} conflicting MACs'.format(len(conflict_macs)))
+            
+            for mac in conflict_macs.values():
+                osrest.neutron_port_delete(auth, mac['neutron_port_id'])
```
Comment (diurnalist) [inline hammers/scripts/conflict_macs.py]: I inlined the `delete` stuff here for simplicity

File: hammers/scripts/conflict_macs.py [hunk 2]
```diff
@@ -77,20 +77,36 @@ def main(argv=None):
...
+
+            if slack:
+                message = 'Fixed Ironic/Neutron MAC conflicts\n{}'.format(
```
Comment (diurnalist) [inline hammers/scripts/conflict_macs.py]: Good rule for logs: log things that happened, not things that are supposedly going to happen. Most logs were like "About to do this..." when it's much more helpful to say "Did this." or "Tried to do this, but failed." Logs can lie, especially when they are talking about hypothetical things. A log could indicate that X is going to happen, but then an error could occur preventing X from even starting.

File: hammers/scripts/lease_stack_reaper.py [hunk 3]
```diff
@@ -206,43 +193,38 @@ def main(argv=None):
...
-                '(Lease stacking restriction violated)'
-                .format(remove_count)
-            )
-            color = '#000000'
-        else:
-            message = ('No leases to delete.')
-            color = '#cccccc'
-        slack.post('lease-stack-reaper', message, color=color)
+    try:
+        users = collect_user_leases(db)
```
Comment (diurnalist) [inline hammers/scripts/lease_stack_reaper.py]: Changed this to return the user/lease structures, then we decide what to do based on `args.action`.

File: hammers/scripts/clean_old_aggregates.py [hunk 4]
```diff
@@ -42,57 +43,53 @@ def aggregates_for_lease(lease):
...
 
-    if report != []:
-      return str_report
-    else:
-      return None
+    return errors, report
 
 
 def main(argv=None):
+    slack = Slackbot(args.slack, script_name='clean-old-aggregates') if args.slack else None
```
Comment (jakecoll) [inline hammers/scripts/clean_old_aggregates.py]: Do we really want one more pep8 violation in hammers just to go down to one line? I'd like refactor hammers at some point to get rid of all these 79>char lines.

Maybe:
`slack = None
if args.slack:
    Slackbot(args.slack, script_name='clean-old-aggregates') `

**Pull request merged**